docs/content/guides/non-argument-deps.mdx

```diff
@@ -0,0 +1,136 @@
+
+---
+
+## Bringing in more stopwords
+
+In the [tutorial](/tutorial), we defined an asset that contains a list of stopwords to filter out common words or prepositions from our word cloud. The English language has a lot of stopwords, so rather than hardcoding them into your asset, you'll bring them in from a separate data source.
+
+You'll augment your asset graph with a new asset that contains the stopwords from a zip file on the internet and create an asset from those stopwords. These stopwords will replace the hardcoded stopwords in your `topstories_word_cloud` asset. By the end of this guide, your asset graph will have two new assets:
+
+- `stopwords_zip` - A zip file on our local filesystem, which contains the stopwords.
+- `stopwords_csv` - A list of stopwords, which contains the unzipped contents of `stopwords_zip`.
+
+### A zip file of stopwords from the internet
+
+Let's start with `stopwords_zip`. We'll add this to the same `assets.py` you wrote in the [tutorial](/tutorial):
+
+```python
+import urllib.request # Addition: new import at the top of `assets.py`
+
+@asset
+def stopwords_zip() -> None:
+    urllib.request.urlretrieve(
+        "https://docs.dagster.io/assets/stopwords.zip",
+        "stopwords.zip",
+    )
+```
+
+The implementation downloads a zip file from the internet to our local filesystem. It looks a lot like the assets we defined earlier in this tutorial, with two differences:
+
+- **It doesn't have a return statement.** Instead of returning an object and letting the I/O manager write it to storage, the function handles writing to storage itself.
+- **It has a `None` annotation for its return type.** This helps Dagster understand what we already know by looking at it - that the function never returns any values.
+
+### An unzipped CSV of stopwords
+
+Let's define an asset that contains the unzipped contents of our zip file:
+
+```python
+import zipfile
+
+@asset(non_argument_deps={"stopwords_zip"})
+def stopwords_csv() -> None:
+    with zipfile.ZipFile("stopwords.zip", "r") as zip_ref:
+        zip_ref.extractall(".")
+```
+
+The implementation extracts our zip file to a CSV file in the same directory. It looks a lot like the assets we defined earlier in this tutorial, with one main difference.
+
+In the assets we've seen so far, we indicate the asset dependencies by including function arguments with the name of the upstream assets. Dagster then invokes an I/O manager to load the asset as a Python object and supply it to our function.
+
+However, in this case, we don't want a Python object for the upstream asset, we just want to work with the file directly. So we instead supply `"stopwords_zip"` to the `non_argument_deps` parameter of the <PyObject module="dagster" object="asset" displayText="@asset"/> decorator. This lets us tell Dagster that the `stopwords` asset depends on the `stopwords_zip` asset, without telling it to load `stopwords_zip` for us.
```

Comment:
```suggestion
However, in this case, we don't want a Python object for the upstream asset, we just want to work with the file directly. So we instead supply `"stopwords_zip"` to the `non_argument_deps` parameter of the <PyObject object="asset" decorator /> decorator. This lets us tell Dagster that the `stopwords` asset depends on the `stopwords_zip` asset, without telling it to load `stopwords_zip` for us.
```

This version of `PyObject` should work as well - much simpler

Reply:
TIL